### PR TITLE
TC_01.015.03 | Freestyle project > Move Project

### PR DESCRIPTION
**Pull request by Lifantsova**
https://trello.com/c/pqy0a1KB/638-tc0101503-freestyle-project-move-project

### CODE REVIEW

**Review by UlyanaVer — APPROVED** [review]

File: src/test/java/school/redrover/FreestyleProject13Test.java [hunk 1]
```diff
@@ -64,22 +65,31 @@ public void testEditExistingDescription() {
...
+        createProject("Freestyle project", PROJECT_NAME);
+        createProject("Folder", NEW_FOLDER);
+
+        getDriver().findElement(By.xpath("//td/a[@href = 'job/" + PROJECT_NAME + "/']")).click();
+        getDriver().findElement(By.xpath("//a[@href = '/job/" + PROJECT_NAME + "/move']")).click();
+
+        Select select = new Select(getDriver().findElement(By.xpath("//select[@name = 'destination']")));
+        select.selectByValue("/" + NEW_FOLDER);
+
+        getDriver().findElement(By.xpath("//button[@name='Submit']")).click();
```
[the remark this answers is not in the document]
Comment (Lifantsova) [reply src/test/java/school/redrover/FreestyleProject13Test.java:88]: Это для красоты))

**Review by AliaksandrMakaranka — APPROVED** [review]

File: src/test/java/school/redrover/FreestyleProject13Test.java [hunk 2]
```diff
@@ -11,36 +12,36 @@ public class FreestyleProject13Test extends BaseTest {
...
-        getDriver().findElement(By.xpath("//input[@name='name']")).sendKeys(PROJECT_NAME);
-        getDriver().findElement(By.xpath("//li[@class='hudson_model_FreeStyleProject']")).click();
-        getDriver().findElement(By.xpath("//button[@type='submit']")).click();
-        getDriver().findElement(By.xpath("//button[@name='Submit']")).click();
+    private void createProject(String itemType, String itemName) {
+        getDriver().findElement(By.xpath("//div[@id='side-panel']//a[contains(@href,'newJob')]")).click();
+        getDriver().findElement(By.xpath("//input[@class='jenkins-input']")).sendKeys(itemName);
+        getDriver().findElement(By.xpath("//span[text()='" + itemType + "']/..")).click();
+        getDriver().findElement(By.xpath("//button[@id='ok-button']")).click();
+        getDriver().findElement(By.xpath("//button[@name = 'Submit']")).click();
```
Comment (EvgenyJavaTest) [inline src/test/java/school/redrover/FreestyleProject13Test.java:20]: 20: This step is redundant. Item was created in the previous step.

File: src/test/java/school/redrover/FreestyleProject13Test.java [hunk 3]
```diff
@@ -11,36 +12,36 @@ public class FreestyleProject13Test extends BaseTest {
...
+        getDriver().findElement(By.cssSelector("#jenkins-name-icon")).click();
```
Comment (EvgenyJavaTest) [inline src/test/java/school/redrover/FreestyleProject13Test.java:21]: 21: Of course, you decide which steps to prescribe in the method. But, strictly speaking, returning to the dashboard does not apply to creating a project. It seems to me that this is not a logical step.

File: src/test/java/school/redrover/FreestyleProject13Test.java [hunk 4]
```diff
@@ -11,36 +12,36 @@ public class FreestyleProject13Test extends BaseTest {
...
+    private void addDescription() {
+        getDriver().findElement(By.xpath("//a[@href='job/FreestyleProject/']")).click();
+        getDriver().findElement(By.id("description-link")).click();
+        getDriver().findElement(By.xpath("//textarea[@name='description']")).sendKeys(DESCRIPTION);
+        getDriver().findElement(By.xpath("//button[@class='jenkins-button jenkins-button--primary ']")).click();
         getDriver().findElement(By.id("jenkins-name-icon")).click();
     }
 
     @Test
     public void testCreateFreestyleProject() {
```
Comment (EvgenyJavaTest) [inline src/test/java/school/redrover/FreestyleProject13Test.java:33]: In this case, this is a positive test of creating a project with a valid name. It makes sense to call this test testCreateFreestyleProject**WithValidName**

File: src/test/java/school/redrover/FreestyleProject13Test.java [hunk 5]
```diff
@@ -64,22 +65,31 @@ public void testEditExistingDescription() {
...
 
-    private void createFolder() {
-        getDriver().findElement(By.xpath("//a[@href='newJob']")).click();
-        getDriver().findElement(By.xpath("//input[@name='name']")).sendKeys(NEW_FOLDER);
-        getDriver().findElement(By.id("j-add-item-type-nested-projects")).click();
-        getDriver().findElement(By.xpath("//button[@type='submit']")).click();
-        getDriver().findElement(By.xpath("//button[@name='Submit']")).click();
-        getDriver().findElement(By.id("jenkins-name-icon")).click();
+    @Test
+    public void testCreateFolder() {
```
Comment (EvgenyJavaTest) [inline src/test/java/school/redrover/FreestyleProject13Test.java:69]: In this case, this is a positive test of creating a folder with a valid name. It makes sense to call this test testCreateFolderWithValidName

**Pull request merged**